lib/python/treadmill_aws/ec2client.py

```diff
@@ -86,9 +97,38 @@ def list_instances(ec2_conn, ids=None, tags=None, hostnames=None, state=None):
 
     for reservation in reservations:
         instances.extend(reservation['Instances'])
+
+    # working around a bug in instance-lifetime filter
+    if spot is True:
+        return list(filter(lambda r: 'InstanceLifecycle' in r, instances))
+    elif spot is False:
+        return list(filter(lambda r: 'InstanceLifecycle' not in r, instances))
+
     return instances
 
 
+def list_spot_requests(ec2_conn):
```

Comment:
If you want to return a complex object when called, that's OK- could you write a formatter to turn the object into a user parse-able format like a table?
treadmill admin aws instance spot list, etc.